src/txn.ts

```diff
@@ -31,13 +31,21 @@ export class Txn {
     private readonly ctx: messages.TxnContext;
     private finished: boolean = false;
     private mutated: boolean = false;
+    private readonly readOnly: boolean = false;
+    private useBestEffort: boolean = false;
     private sequencingProp: messages.LinRead.SequencingMap[keyof messages.LinRead.SequencingMap];
 
-    constructor(dc: DgraphClient) {
+    constructor(dc: DgraphClient, txnOpts?: { readOnly: boolean } | null) {
```

Comment:
tslint complains about "| null" part so we stopped using it everywhere else.
`txnOptions?: { ...accepted options... }` is good enough.
Also when you do `txnOpts !== undefined ? txnOpts.readOnly` passing null would actually cause a crash.

Reply:
Normally I would use `txnOpts ?` but the linting rules setup complain about that as well.

Reply:
can you please paste a tslint error here?
I get no errors when I add 
`constructor(dc: DgraphClient, txnOptions: { readOnly?: boolean; bestEffort?: boolean })`
to my local repo

Reply:
That's because `txnOptions` is not optional in your example

Reply:
I updated the code with new style.